File: @LV2/2.8/backspace/backspace.go
// +----------+
// | TOO SLOW |
// +----------+
package main

import "fmt"

func main() {
	var line string
	var cnt int
	out := make([]byte, 1000010)

	fmt.Scanln(&line)

	for _, ch := range line {
		if ch == '<' {
			if cnt > 0 {
				cnt--
			}
		} else {
			out[cnt] = byte(ch)
			cnt++
		}
	}
	fmt.Println(string(out[0:cnt]))
}

File: @LV2/2.8/backspace/backspace.rb
# +----------+
# | TOO SLOW |
# +----------+
line = gets
out = [''] * 1_000_000
cnt = 0

line.each_char do |ch|
  if ch == '<'
    cnt -= 1 if cnt > 0
  else
    out[cnt] = ch
    cnt += 1
  end
end

puts out[0...cnt].join('')
